Mathlib/AlgebraicGeometry/Sites/BigAffineZariski.lean

```diff
@@ -0,0 +1,311 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import Mathlib.AlgebraicGeometry.PullbackCarrier
+import Mathlib.AlgebraicGeometry.Sites.BigZariski
+import Mathlib.CategoryTheory.Sites.DenseSubsite.InducedTopology
+import Mathlib.CategoryTheory.Sites.Finite
+import Mathlib.RingTheory.RingHom.OpenImmersion
+
+/-!
+# The Big Zariski Site on the Opposite Category of Commutative Rings
+
+In this file we define the Zariski topology on `CommRingCatᵒᵖ`, noting that this category is
+equivalent to the category of affine schemes.
+
+## Main Definitions
+
+- `CommRingCat.zariskiTopology`: The Zariski topology on `CommRingCatᵒᵖ`, constructed using the
+  induced topology from the Zariski topology on the category of schemes via the functor
+  `AlgebraicGeometry.Scheme.Spec` : `CommRingCatᵒᵖ ⥤ Scheme`.
+- `CommRingCat.zariskiPretopology`: The Zariski pretopology on `CommRingCatᵒᵖ`, consisting of
+  presieves that are finite, jointly surjective, and standard open immersions.
+
+## Main Results
+
+- `zariskiTopology_eq_toGrothendieck_zariskiPretopology`: The Zariski topology on `CommRingCatᵒᵖ` is
+  equal to the Grothendieck topology generated by the standard pretopology.
+- `sheafEquiv`: An equivalence between the category of sheaves on the Zariski site of
+  `CommRingCatᵒᵖ` and on the Zariski site of `Scheme`. In other words, a sheaf on `CommRingCatᵒᵖ`
+  can be extended uniquely to `Scheme`.
+
+-/
+
+universe u₀ v v₁ v₂ v₃ u u₁ u₂
+
+open CategoryTheory Functor Opposite Category Limits CommRingCat
+open AlgebraicGeometry AffineScheme Scheme TensorProduct PrimeSpectrum
+
+namespace AlgebraicGeometry
+
+namespace Scheme
+
+variable {X : Scheme.{u}} {P : MorphismProperty Scheme.{u}}
+
+/-- An open immersion `u : U ⟶ Spec R` is covered by `Spec R[1/f]`. In other words, for every
+`p : U`, there is `f : R` such that `p ∈ D(f) ⊆ U`, i.e. such that `Spec R[1/f] ⟶ Spec R`
+factors through `u`. -/
+lemma Hom.exists_factor
+    {R : Type u} [CommRing R] {U : Scheme.{u}} (u : U ⟶ Spec (.of R)) [IsOpenImmersion u] (p : U) :
+    ∃ f : R, ∃ g : Spec (.of (Localization.Away f)) ⟶ U,
+      PrimeSpectrum.basicOpen f ≤ u.opensRange ∧
+      IsOpenImmersion g ∧
+      g ≫ u = Spec.map (ofHom (algebraMap R (Localization.Away f))) ∧
+      p ∈ Set.range g.base := by
```

Comment:
This should probably be moved to the file where `IsOpenImmersion` is defined.

Reply:
done